ERCS/erc-7690.md

```diff
@@ -0,0 +1,129 @@
+---
+eip: 7690
+title: Union Lock
+description: Union Lock based on TSTORE/TLOAD can avoid flash loan attacks.
+author: Elon Lee (@1999321)
+discussions-to: https://ethereum-magicians.org/t/union-lock-based-on-tstore-tload-can-avoid-flash-loan-attacks/19676
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-16
+requires: 8000
+---
+
+## Abstract
+
+The feature of Union Lock can query the number of calls of other contracts and the number of calls of other contract functions in the same Ethereum transaction, thereby realizing the cross-contract locking function.
+
+Union Lock is based on  TSTORE/TLOAD, it can make DApps avoid flash loan attacks.
+
+## Motivation
+
+Since the losses caused by flash loan attacks are as high as tens of millions of dollars, it would be meaningful if a design could be provided to enable DApps to avoid flash loan attacks.
+
+## Specification
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code in the proposal file must be licensed under `CC0-1.0`. If that isn't possible, you can put MIT licensed stuff in the `assets` directory.

Reply:
change it to "// CC0 1.0 Universal (CC0 1.0) Public Domain Dedication"